### DISCUSSION TITLE
Can I use S3.getObject().createReadStream() This Stream encrypt & decrypt ??

### DISCUSSION THREAD

**Discussion by hanoj-budime** [openpgpjs/openpgpjs#1472] [Q&A]
- OpenPGP.js version: 5.1.0 ( latest )
- Affected platform (Browser or Node.js version): 8+

Using Stream can we do encrypt & decrypt data ?

`AWS.S3().getObject().createReadStream();` 

This is the following article ( https://gist.github.com/rich-nahra/b3bea178e299c30a4836d04cef655d5a ) openpgp@4.10 Version
there is worked.

But in openPGP@latest version iam not able do it with this way ( S3.getObject().createReadStream() );

```
const reader = s3.getObject({ Bucket, Key: `test` }).createReadStream();
const decrypted = await openpgp.encrypt({
    message: (await openpgp.message.fromText(reader, 'node')),
    publicKeys: keys.publicKey.keys,
});
const cipherText = decrypted.data;
```

````
const reader = s3.getObject({ Bucket, Key: EncryptedDoc }).createReadStream();
const decrypted = await openpgp.decrypt({
      message: (await openpgp.message.readArmored(reader, 'node')),
      privateKeys: keys.privateKey.keys,
  });
```

**Comment by larabr** [comment]
Hi @HanojHanu , the API changed substantially from v4 to v5. Have a look at the README or [here](https://github.com/openpgpjs/openpgpjs/wiki/V5-Changelog#high-level-api-changes) to see how to port your code from v4 to v5.

**Reply by hanoj-budime** [reply]
So, Can you help with Version 5, Syntax to my snippet code

This is V4 using code
```
const reader = s3.getObject({ Bucket, Key: `test` }).createReadStream();
const decrypted = await openpgp.encrypt({
    message: (await openpgp.message.fromText(reader, 'node')),
    publicKeys: keys.publicKey.keys,
});
const cipherText = decrypted.data;

const reader = s3.getObject({ Bucket, Key: EncryptedDoc }).createReadStream();
const decrypted = await openpgp.decrypt({
      message: (await openpgp.message.readArmored(reader, 'node')),
      privateKeys: keys.privateKey.keys,
  });

```

V5 code will be ??
```
// code here
```

**Reply by hanoj-budime** [reply]
I tried but i got error with Version 5, So can you help me with syntax.

```
Error: Misformed armored text
    at c:\WorkSpace\try\OpenPGP\node_modules\openpgp\dist\node\openpgp.min.js:2:26297
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Reply by larabr** [reply]
If you share your v5 code I can point out the problem
